kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -304,24 +304,58 @@ impl<'tcx> GotocCtx<'tcx> {
             }
             (Scalar::Int(_), ty::Adt(adt, subst)) => {
                 if adt.is_struct() || adt.is_union() {
-                    // in this case, we must have a one variant ADT. there are two cases
-                    let variant = &adt.variants().raw[0];
-                    // if there is no field, then it's just a ZST
+                    // In this case, we must have a one variant ADT. There are two cases.
+                    let variant = adt.non_enum_variant();
+                    let overall_type = self.codegen_ty(ty);
+                    // If there is no field, then it's just a ZST
                     if variant.fields.is_empty() {
                         if adt.is_struct() {
-                            let overall_t = self.codegen_ty(ty);
-                            Expr::struct_expr_from_values(overall_t, vec![], &self.symbol_table)
+                            Expr::struct_expr_from_values(overall_type, vec![], &self.symbol_table)
                         } else {
                             unimplemented!()
                         }
                     } else {
-                        // otherwise, there is just one field, which is stored as the scalar data
-                        let field = &variant.fields[0usize.into()];
-                        let fty = field.ty(self.tcx, subst);
-
-                        let overall_t = self.codegen_ty(ty);
+                        // Otherwise, there is at least one field. Again, there are two cases.
+                        let overall_type = self.codegen_ty(ty);
                         if adt.is_struct() {
-                            self.codegen_single_variant_single_field(s, span, overall_t, fty)
+                            if variant.fields.len() == 1 {
```

Comment:
Can this branch and the next branch of the if-then-else be merged ? it seems like the single  non-zst field case is a special case of n-fields with a single non-zst.

Reply:
Yes, it's done. I've even gone further and removed the `adt.is_struct()` condition by restricting the outer one only to structs.